packages/kernel/src/kernels.ts

```diff
@@ -75,7 +75,14 @@ export class Kernels implements IKernels {
           } else {
             return;
           }
-          void processMsg(msg);
+
+          // TODO Find a better solution for this?
+          // input-reply is asynchrone, must not be processed like other messages
+          if (msg.header.msg_type === 'input_reply') {
+            kernel.handleMessage(msg);
+          } else {
+            void processMsg(msg);
+          }
```

Comment:
@jtpio the mutex is preventing us from getting the `input_reply` while the `execute_request` is being processed, so we need to do this. We might want to find a better way to handle this case.

Reply:
Looks like it should be fine like that yes.

For now the mutex is merely there so the kernel processes one message at a time. And have the queue of messages processed in order.

Reply:
I am resolving this since it's okay for now. Feel free to "un-resolve" it again if needed.